readthedocs/search/api.py

```diff
@@ -35,14 +37,71 @@ def get_link(self, obj):
     def get_highlight(self, obj):
         highlight = getattr(obj.meta, 'highlight', None)
         if highlight:
-            if hasattr(highlight, 'content'):
-                # Change results to turn newlines in highlight into periods
-                # https://github.com/rtfd/readthedocs.org/issues/5168
-                highlight.content = [result.replace('\n', '. ') for result in highlight.content]
-            ret = highlight.to_dict()
-            log.debug('API Search highlight: %s', pformat(ret))
+            ret = self._get_clean_highlight_dict(highlight.to_dict())
+            log.debug('API Search highlight [Page title]: %s', pformat(ret))
             return ret
 
+    def get_inner_hits(self, obj):
+        inner_hits = getattr(obj.meta, 'inner_hits', None)
+        if inner_hits:
+            res = defaultdict(list)
+
+            # add sections data to the search results
+            sections = getattr(inner_hits, 'sections', None)
+            for hit in sections.hits:
+                section_highlight = self._get_clean_highlight_dict(
+                    hit.highlight.to_dict()
+                )
+
+                log.debug('API Search highlight [Page sections]: %s', pformat(section_highlight))
```

Comment:
Are the `logs` necessary here.
We might end up having too much of logs.

Reply:
Yes we should definitely log, and continue to add logging where it makes sense as well.